docs/NSR.md

```diff
@@ -347,11 +340,11 @@ The CA MUST enforce the use of Multi-Party Control for physical access to any Ro
 
 The CA SHOULD ensure passwords used as authentication credentials for accounts on CA Infrastructure, Network Equipment, or Workstations are generated and managed in accordance with NIST 800-63B Appendix A.
 
-The CA SHALL NOT require periodic password changes with a period less than two (2) years.
+The CA SHOULD NOT require periodic password changes with a period less than two (2) years.
 
-The CA MUST ensure passwords used as authentication credentials for accounts on CA Infrastructure have a minimum of twelve (12) characters.
+The CA MUST require that passwords used as authentication credentials for accounts on CA Infrastructure, Network Equipment or Workstations have a minimum of twelve (12) characters.
 
-The CA MUST ensure passwords used as authentication credentials for accounts on Network Equipment or Workstations have a minimum of eight (8) characters.
+The CA SHOULD ensure passwords used as authentication credentials for accounts on CA Infrastructure, Network Equipment, or Workstations are programmatically generated and stored in a credential manager. Access to credentials stored in a credential manager MUST be limited to personnel based on the Principle of Least Privilege.
```

Comment:
I am not sure this really makes a lot of sense. First, storing credentials for workstations ("a phone, tablet, or desktop or laptop computer") is not realistic or practical. Consulting a password manager before _entering_ a password on one of these devices is simply not realistic for how anyone performs this kind of activity; especially since these devices would be those that a password manager would be consulted/interacted with.

But even going beyond that, we have to assume that passwords used for the listed systems are going to be extremely sensitive; in many instances these passwords will be credentials of last resort (i.e. when better or more coordinated ways of authenticating fail - think "root password" - or the equipment is somehow fundamental and can't be integrated with other authentication mechanisms - think HSM administrative credentials). Concentrating them in a password manager brings said password manager close to a single point of failure.

I do not mean to suggest that such credentials should instead be managed on post-it notes, but at the same time I would not necessarily rule out that storing them on paper in an appropriate kind of vault with appropriately controlled access could not be a proper way of dealing with them.

At the same time,

> are [...] stored in a credential manager. Access to credentials stored in a credential manager MUST be limited to personnel based on the Principle of Least Privilege.

would most certainly give the impression that regular consumer or even enterprise level password managers would not be acceptable but what would be in fact indicated by this provision. Many of these password manager products offer a level of differentiated access to shared credentials, which would further solidify this interpretation.

Obviously, such password managers are not free from failure, as we have seen by successful and nearly successful attacks on them.

Some available password managers might be a good fit, or might offer modes of operation that are a good fit, for a CA's and the WebPKI ecosystem's security needs regarding management of passwords. Some of them, and certainly in some modes of operation, willl not be. I believe we must avoid adding a provision that prompts an implementation which would not be aligned with the security needs.

Reply:
I think you make some valid points, especially regarding the sensitivity of these passwords. I do not agree that using some ad-hoc solution rather than using well-vetted commercial or open source solutions with good encryption would be better. 
I've seen this level of secrets, especially shared ones, written on paper and stored in tamper bags in safes. This is not less onerous than accessing a credential manager or decrypting stored secrets for rarely used passwords. This wording still allows for other solutions while strongly encouraging the use of strong technical controls.

Reply:
In light of the changes to remove 350-352, and @tobij's dislike for including the reference to credential managers, perhaps we could actually delete this line as well. 
 I think there is value to the second part regarding shared credentials and, while I personally like encouraging the use of credential managers and programmatically generated passwords, the reference to NIST 800-63B Appendix A technically pulls those in. Maybe this should become something like:
> Access to shared credentials MUST:
>  * be limited to personnel based on the Principle of Least Privilege; and
>  * comply with section 2.2.1.2.

Reply:
Let's mention the specific version of NIST 800-63B